server/index.js

```diff
@@ -0,0 +1,38 @@
+import express from 'express';
+import cors from 'cors';
+import sqlite3 from 'sqlite3'; // Import sqlite3
+import blogControllerRoutes from './Controllers/blogControllers.js';
+
+const app = express();
+const PORT = 3030;
+
+// Middleware
+app.use(cors());
+app.use(express.json());
+
+// Initialize SQLite3 database
+const db = new sqlite3.Database('deurdu.db', (err) => {
+  if (err) {
+    console.error('Error connecting to the database:', err);
+  } else {
+    console.log('Connected to the SQLite database');
+  }
+});
+
+// Make the database connection available to routes
+app.use((req, res, next) => {
+  req.db = db; // Attach the database connection to the request object
```

Comment:
if you connected `db` to request object, you don't need to initialize the database again in https://github.com/fsheikh/sample_code/pull/11/files#diff-9ac2ac1d104a8ac8f861c556d1082ec1051171365a02a5e5355d49265b437177R7

Reply:
Yes, exactly!

By attaching the database to `req` using:

```bash
app.use((req, res, next) => {
  req.db = db;
  next();
});
```
You make the database connection globally available to all routes.